internal/util/debug/archive.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package debug provides debug facilities.
+package debug
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path/filepath"
+)
+
+type archiveUrlList []url.URL
+
+// archiveHandler - is a handler for creating the archive containing all the debug and metrics info.
+func archiveHandler(rw http.ResponseWriter, req *http.Request) {
+	var (
+		scheme  string
+		urlList archiveUrlList
+	)
+
+	urlList = make(archiveUrlList, 0)
+
+	u := new(url.URL)
+	u.Path = metricsPath
+	u.Host = req.Host
+
+	if req.URL.Scheme == "" {
+		scheme = "http"
+	}
+
+	u.Scheme = scheme
+	urlList = append(urlList, *u)
```

Comment:
I'm sorry, but that's a very complicated way. Just use `u := &url.URL{...}`

Reply:
simplified the usage.

Reply:
Finding : 
Linter shows a message as below for usage of &url . That was the reason I made it like that earlier.

./internal/util/debug/archive.go:35:7: zero value ptr alloc: use new(T) for *T allocation
./internal/util/debug/archive.go:46:6: zero value ptr alloc: use new(T) for *T allocation